File: BaekJoon/1912_연속핪.java
import java.util.Scanner;
public final class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }
        int max = -1001; int tmp = -1001; int turn = 0;
        for (int i = 0; i < arr.length; i++) {
            if (tmp > arr[i] || tmp == -1001 || tmp < tmp+arr[i]) {
                if (tmp == -1001) tmp = 0;
                tmp += arr[i];
            }
            if (tmp <= arr[i]) {
                tmp = 0;
                tmp += arr[i];
            }
            if (tmp > max) max = tmp;
        }
        System.out.println(max);
    }
}

File: BaekJoon/1912_연속핪.kt
import java.util.*
fun main() {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val arr = IntArray(n)
    for (i in arr.indices) {
        arr[i] = sc.nextInt()
    }
    var max = -1001
    var tmp = -1001
    val turn = 0
    for (i in arr.indices) {
        if (tmp > arr[i] || tmp == -1001 || tmp < tmp + arr[i]) {
            if (tmp == -1001) tmp = 0
            tmp += arr[i]
        }
        if (tmp <= arr[i]) {
            tmp = 0
            tmp += arr[i]
        }
        if (tmp > max) max = tmp
    }
    println(max)
}
